standalone/test/integration_tests/api/integrationtest_scanner_api.cpp

```diff
@@ -149,30 +155,94 @@ void ScannerAPITests::prepareScannerMockStartReply()
   }
 }
 
-std::unique_ptr<util::Barrier> ScannerAPITests::prepareStrictMockStartRequestBarrier(const ScannerConfiguration& config)
+std::unique_ptr<util::Barrier> ScannerAPITests::prepareStartRequestBarrier(const ScannerConfiguration& config)
 {
   const data_conversion_layer::start_request::Message start_req(config);
   auto start_req_received_barrier = std::make_unique<util::Barrier>();
-  EXPECT_CALL(*strict_scanner_mock_, receiveControlMsg(_, data_conversion_layer::start_request::serialize(start_req)))
-      .WillOnce(OpenBarrier(start_req_received_barrier.get()));
+
+  if (strict_scanner_mock_)
+  {
+    EXPECT_CALL(*strict_scanner_mock_, receiveControlMsg(_, data_conversion_layer::start_request::serialize(start_req)))
+        .WillOnce(OpenBarrier(start_req_received_barrier.get()));
+  }
+  else
+  {
+    ON_CALL(*nice_scanner_mock_, receiveControlMsg(_, data_conversion_layer::start_request::serialize(start_req)))
+        .WillByDefault(OpenBarrier(start_req_received_barrier.get()));
+  }
   return start_req_received_barrier;
 }
 
+std::unique_ptr<util::Barrier> ScannerAPITests::prepareMonitoringFrameBarrier(
+    const std::vector<data_conversion_layer::monitoring_frame::Message>& msgs)
+{
+  const auto timestamp{ util::getCurrentTime() };
+  const auto scan{ createReferenceScan(msgs, timestamp) };
+
+  auto monitoring_frame_barrier = std::make_unique<util::Barrier>();
+
+  EXPECT_CALL(user_callbacks_,
+              LaserScanCallback(AllOf(ScanDataEqual(scan), TimestampInExpectedTimeframe(scan, timestamp))))
+      .WillOnce(OpenBarrier(monitoring_frame_barrier.get()));
+  return monitoring_frame_barrier;
+}
+
+void ScannerAPITests::startScanner()
+{
+  prepareScannerMockStartReply();
+  if (strict_scanner_mock_)
+  {
+    strict_scanner_mock_->startListeningForControlMsg();
+  }
+  else
+  {
+    nice_scanner_mock_->startListeningForControlMsg();
+  }
+  const auto start_future = scanner_->start();
+  expectIsDone("Scanner::start()", start_future);
+}
+
+void ScannerAPITests::expectNotDoneYet(const std::string& method_name, const std::future<void>& future)
+{
+  EXPECT_EQ(future.wait_for(FUTURE_WAIT_TIMEOUT), std::future_status::timeout) << method_name << " finished to early";
+}
+
+void ScannerAPITests::expectIsDone(const std::string& method_name, const std::future<void>& future)
```

Comment:
No a fan. Returning a future is part of the API. Using exactly this future and its functions in the tests seems fine to me.

Reply:
Also passing the string just to be able to print it is not optimal. I immediatly assumed there is something more done with it.

Reply:
You are totally right about the string. The goal was to make the intent of the expectation more clear, because expect future.wait_for() to be equal to something does not tell me what its about right away. I had to read the result string to get that. Introduced a define now that hopefully increases readability without passing a log string to a method

Reply:
You are right about the string.
My intention was to ease the readability. I was always trying to understand the equation at specially on the = timeout part and then defaulted on reading the message on the line end.
Introduced a define now, which hopefully satisfies you complains and helps to read the line faster.